--- origsrc/kdemultimedia-3.5.9/akode_artsplugin/Makefile.am	2006-07-22 03:10:53.000000000 -0500
+++ src/kdemultimedia-3.5.9/akode_artsplugin/Makefile.am	2008-04-29 20:56:08.109375000 -0500
@@ -8,7 +8,7 @@
         akodeMPCPlayObject_impl.cpp akodeMPEGPlayObject_impl.cpp \
         akodeFFMPEGPlayObject_impl.cpp akodeXiphPlayObject_impl.cpp \
         akodeVorbisStreamPlayObject_impl.cpp akodeSpeexStreamPlayObject_impl.cpp
-libarts_akode_la_LDFLAGS = $(all_libraries) -module -no-undefined
+libarts_akode_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 libarts_akode_la_LIBADD =  $(akode_libs) -lkmedia2_idl -lsoundserver_idl -lartsflow
 libarts_akode_la_METASOURCES = AUTO
 libarts_akode_la_COMPILE_FIRST = akodearts.h
--- origsrc/kdemultimedia-3.5.9/arts/gui/kde/Makefile.am	2005-09-10 03:19:46.000000000 -0500
+++ src/kdemultimedia-3.5.9/arts/gui/kde/Makefile.am	2008-04-29 20:56:08.156250000 -0500
@@ -27,7 +27,7 @@
 
 dbtest_SOURCES = dbtest.cpp
 dbtest_LDADD = -lqtmcop -lkdecore -lartsflow \
-	$(top_builddir)/arts/gui/kde/libartsgui_kde.la
+	libartsgui_kde.la
 dbtest_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 libartsgui_kde_la_METASOURCES = AUTO
--- origsrc/kdemultimedia-3.5.9/arts/modules/Makefile.am	2005-09-10 03:19:48.000000000 -0500
+++ src/kdemultimedia-3.5.9/arts/modules/Makefile.am	2008-04-29 20:56:08.187500000 -0500
@@ -28,14 +28,7 @@
 libartsmodules_la_SOURCES = artsmodules.cc
 
 libartsmodules_la_LIBADD = \
-   $(top_builddir)/arts/runtime/libartsbuilder.la \
-   $(top_builddir)/arts/midi/libartsmidi_idl.la \
-   $(top_builddir)/arts/gui/common/libartsgui_idl.la \
-   $(top_builddir)/arts/modules/synth/libartsmodulessynth.la \
-   $(top_builddir)/arts/modules/common/libartsmodulescommon.la \
-   $(top_builddir)/arts/modules/effects/libartsmoduleseffects.la \
-   $(top_builddir)/arts/modules/mixers/libartsmodulesmixers.la \
-   -lartsflow -lmcop $(LIB_KDECORE) $(LIBDL)
+   -lartsflow -lmcop $(LIBDL)
 libartsmodules_la_LDFLAGS = $(all_libraries) -L$(arts_libraries) \
    -no-undefined
 
--- origsrc/kdemultimedia-3.5.9/arts/tools/Makefile.am	2005-09-10 03:19:45.000000000 -0500
+++ src/kdemultimedia-3.5.9/arts/tools/Makefile.am	2008-04-29 22:36:43.750000000 -0500
@@ -32,12 +32,15 @@
 SOMINOR = 0
 SOSUBMINOR = 0
 
-libartscontrolapplet_la_LDFLAGS = $(all_libraries) -version-info $(SOMAJOR):$(SOMINOR):$(SOSUBMINOR) -no-undefined -module
+libartscontrolapplet_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 libartscontrolsupport_la_LDFLAGS = $(all_libraries) -version-info $(SOMAJOR):$(SOMINOR):$(SOSUBMINOR) -no-undefined
 
 libartscontrolapplet_la_LIBADD = \
+	$(top_builddir)/arts/gui/common/libartsgui_idl.la \
+	$(top_builddir)/arts/gui/kde/libartsgui_kde.la \
+	$(top_builddir)/arts/modules/effects/libartsmoduleseffects.la \
 	-lartsflow -lartsflow_idl -lmcop -lqtmcop $(LIBDL) -lsoundserver_idl -lartskde \
-	$(LIB_KDEUI) ./libartscontrolsupport.la
+	$(LIB_KDEUI) libartscontrolsupport.la
 
 # Arnolds version with dynamic-linking for testing.
 #
@@ -53,12 +56,14 @@
 	$(top_builddir)/arts/gui/common/libartsgui_idl.la \
 	$(top_builddir)/arts/gui/kde/libartsgui_kde.la \
 	$(top_builddir)/arts/modules/libartsmodules.la \
+	$(top_builddir)/arts/modules/common/libartsmodulescommon.la \
 	$(LIB_KDEUI)
 
 bin_PROGRAMS = artscontrol
 artscontrol_SOURCES = main.cpp levelmeters.cpp
 artscontrol_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 artscontrol_LDADD = libartscontrolsupport.la \
+	$(top_builddir)/arts/modules/effects/libartsmoduleseffects.la \
 	$(LIB_KDEUI) $(LIB_KSYCOCA) $(LIB_KFILE) \
 	-lqtmcop -lartsflow_idl -lsoundserver_idl -lartskde
 
--- origsrc/kdemultimedia-3.5.9/audiofile_artsplugin/Makefile.am	2006-07-22 03:10:54.000000000 -0500
+++ src/kdemultimedia-3.5.9/audiofile_artsplugin/Makefile.am	2008-04-29 20:56:08.296875000 -0500
@@ -5,7 +5,7 @@
 lib_LTLIBRARIES = libarts_audiofile.la
 libarts_audiofile_la_COMPILE_FIRST = audiofilearts.h
 libarts_audiofile_la_SOURCES = audiofilearts.cc audiofilePlayObjectI.cpp
-libarts_audiofile_la_LDFLAGS = $(all_libraries) -module -no-undefined
+libarts_audiofile_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 libarts_audiofile_la_LIBADD =  -lkmedia2_idl -lsoundserver_idl -lartsflow -laudiofile
 libarts_audiofile_la_METASOURCES = AUTO
 
--- origsrc/kdemultimedia-3.5.9/juk/Makefile.am	2007-10-08 04:48:55.000000000 -0500
+++ src/kdemultimedia-3.5.9/juk/Makefile.am	2008-04-29 20:56:08.343750000 -0500
@@ -74,7 +74,7 @@
 
 tagguessertest_SOURCES = tagguessertest.cpp tagguesser.cpp
 
-INCLUDES= $(all_includes) $(taglib_includes) $(akode_includes) $(GST_CFLAGS) $(ARTS_CFLAGS)
+INCLUDES= $(all_includes) $(taglib_includes) $(akode_includes) $(GST_CFLAGS) -I$(arts_includes)
 
 ##################################################
 # check to see if MusicBrainz is available
--- origsrc/kdemultimedia-3.5.9/kfile-plugins/mpeg/kfile_mpeg.cpp	2007-01-15 05:22:58.000000000 -0600
+++ src/kdemultimedia-3.5.9/kfile-plugins/mpeg/kfile_mpeg.cpp	2008-04-29 20:56:08.390625000 -0500
@@ -40,7 +40,9 @@
 
 typedef unsigned char uint8_t;
 typedef unsigned short uint16_t;
+#ifndef __CYGWIN__
 typedef unsigned int uint32_t;
+#endif
 
 typedef KGenericFactory<KMpegPlugin> MpegFactory;
 
--- origsrc/kdemultimedia-3.5.9/kfile-plugins/sid/Makefile.am	2005-09-10 03:20:07.000000000 -0500
+++ src/kdemultimedia-3.5.9/kfile-plugins/sid/Makefile.am	2008-04-29 20:56:08.406250000 -0500
@@ -10,7 +10,7 @@
 
 kfile_sid_la_SOURCES = kfile_sid.cpp
 kfile_sid_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-kfile_sid_la_LIBADD = $(LIB_KIO)
+kfile_sid_la_LIBADD = $(LIB_KIO) $(taglib_libs)
 
 # let automoc handle all of the meta source files (moc)
 METASOURCES = AUTO
--- origsrc/kdemultimedia-3.5.9/kmid/Makefile.am	2005-09-10 03:20:00.000000000 -0500
+++ src/kdemultimedia-3.5.9/kmid/Makefile.am	2008-04-29 20:56:08.453125000 -0500
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = libkmidpart.la
 lib_LTLIBRARIES = libkmidlib.la
 
-libkmidlib_la_LDFLAGS = $(all_libraries)
+libkmidlib_la_LDFLAGS = $(all_libraries) -no-undefined
 libkmidlib_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) -lkmid
 
 libkmidpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
--- origsrc/kdemultimedia-3.5.9/kmix/Makefile.am	2005-09-10 03:20:07.000000000 -0500
+++ src/kdemultimedia-3.5.9/kmix/Makefile.am	2008-04-29 20:56:08.484375000 -0500
@@ -25,14 +25,14 @@
 	volume.cpp kledbutton.cpp verticaltext.cpp mixerIface.skel \
 	kmixtoolbox.cpp mixertoolbox.cpp dialogviewconfiguration.cpp KMixApp.cpp dialogselectmaster.cpp
 
-kmix_la_LIBADD = $(LIB_KDEUI) $(LIB_KUTILS) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND)
+kmix_la_LIBADD = $(LIB_KDEUI) $(LIB_KUTILS) $(LIB_KDECORE) $(LIB_DCOP) $(LIB_QT) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND)
 kmix_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN) -avoid-version
 
 kmixctrl_la_SOURCES = kmixctrl.cpp mixer.cpp mixset.cpp mixdevice.cpp volume.cpp mixerIface.skel \
 	mixertoolbox.cpp mixer_backend.cpp
 
 kmixctrl_la_LIBADD = $(LIB_KDECORE) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND)
-kmixctrl_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kmixctrl_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
 kmix_panelapplet_la_SOURCES = kmixapplet.cpp \
 	viewbase.cpp viewapplet.cpp \
--- origsrc/kdemultimedia-3.5.9/krec/Makefile.am	2005-09-10 03:19:58.000000000 -0500
+++ src/kdemultimedia-3.5.9/krec/Makefile.am	2008-04-29 23:10:06.843750000 -0500
@@ -47,6 +47,7 @@
 	lib_krec_common.la \
 	$(top_builddir)/arts/gui/kde/libartsgui_kde.la \
 	$(top_builddir)/arts/tools/libartscontrolsupport.la \
+	$(top_builddir)/arts/modules/effects/libartsmoduleseffects.la \
 	-lartskde $(LIBDL) $(LIB_KIO) $(LIB_KDEUI) $(LIB_KUTILS)
 
 kcm_krec_la_SOURCES = krecconfigure.cpp
--- origsrc/kdemultimedia-3.5.9/kscd/configure.in.in	2005-09-10 03:19:57.000000000 -0500
+++ src/kdemultimedia-3.5.9/kscd/configure.in.in	2008-04-29 20:56:08.562500000 -0500
@@ -55,7 +55,7 @@
   *-*-*freebsd*) kscd_compile=yes;;
   *-*-*netbsd*) kscd_compile=yes;;
   *-*-*openbsd*) kscd_compile=yes;;
-  *) kscd_compile=yes;;
+  *) kscd_compile=no;;
 esac
 AC_MSG_RESULT($kscd_compile)
 if test "$kscd_compile" = "no"; then
--- origsrc/kdemultimedia-3.5.9/libkcddb/Makefile.am	2005-09-10 03:19:31.000000000 -0500
+++ src/kdemultimedia-3.5.9/libkcddb/Makefile.am	2008-04-29 20:56:08.609375000 -0500
@@ -13,7 +13,7 @@
     synchttpsubmit.cpp cdinfodialogbase.ui categories.cpp genres.cpp \
     cdinfoencodingwidget.cpp cdinfoencodingwidgetbase.ui
 
-libkcddb_la_LDFLAGS = $(all_libraries) -version-info 1:0:0
+libkcddb_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 libkcddb_la_LIBADD  = $(LIB_KDECORE) $(LIB_KIO)
 
 METASOURCES = AUTO
--- origsrc/kdemultimedia-3.5.9/noatun/app/Makefile.am	2005-09-10 03:19:31.000000000 -0500
+++ src/kdemultimedia-3.5.9/noatun/app/Makefile.am	2008-04-29 20:56:08.656250000 -0500
@@ -2,7 +2,7 @@
 
 kdeinit_LTLIBRARIES = noatun.la
 noatun_la_SOURCES = main.cpp
-noatun_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module -avoid-version
+noatun_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 noatun_la_LIBADD = -lqtmcop -lkmedia2_idl -lsoundserver_idl $(top_builddir)/noatun/library/libnoatun.la
 
 bin_PROGRAMS =
--- origsrc/kdemultimedia-3.5.9/noatun/library/Makefile.am	2005-09-10 03:19:34.000000000 -0500
+++ src/kdemultimedia-3.5.9/noatun/library/Makefile.am	2008-04-29 20:56:08.765625000 -0500
@@ -26,7 +26,7 @@
 	ksaver.h
 
 libnoatun_la_LDFLAGS     = \
-		-version-info 3:0:2 $(KDE_RPATH) $(all_libraries)
+		-version-info 3:0:2 $(KDE_RPATH) $(all_libraries) -no-undefined
 libnoatun_la_LIBADD      = $(top_builddir)/arts/gui/common/libartsgui.la \
 	$(top_builddir)/arts/gui/kde/libartsgui_kde.la $(top_builddir)/arts/modules/libartsmodules.la \
 	-lkio -lqtmcop -lkmedia2_idl $(top_builddir)/noatun/library/noatunarts/libnoatunarts.la \
@@ -43,7 +43,7 @@
 gentable_SOURCES = gentable.cpp
 gentable_LDFLAGS = $(KDE_EXTRA_RPATH)
 
-magictable: gentable
+magictable: gentable$(EXEEXT)
 	$(top_builddir)/noatun/library/gentable > magictable
 
 magictabledir = $(kde_datadir)/noatun
--- origsrc/kdemultimedia-3.5.9/noatun/library/noatuntags/Makefile.am	2005-09-10 03:19:33.000000000 -0500
+++ src/kdemultimedia-3.5.9/noatun/library/noatuntags/Makefile.am	2008-04-29 20:56:08.875000000 -0500
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = libnoatuntags.la
 
 libnoatuntags_la_SOURCES = tags.cpp
-libnoatuntags_la_LDFLAGS = -version-info 3:0:2 $(all_libraries)
+libnoatuntags_la_LDFLAGS = -version-info 3:0:2 $(all_libraries) -no-undefined
 libnoatuntags_la_LIBADD = ../libnoatun.la
 
 libnoatuntags_la_METASOURCES = AUTO
--- origsrc/kdemultimedia-3.5.9/oggvorbis_artsplugin/configure.in.in	2005-09-10 03:19:44.000000000 -0500
+++ src/kdemultimedia-3.5.9/oggvorbis_artsplugin/configure.in.in	2008-04-29 20:56:08.906250000 -0500
@@ -1,4 +1,4 @@
-if test "$kde_mpeglib_compiles" = "yes"
+if test "x$build_arts" = "xno"
 then
 DO_NOT_COMPILE="$DO_NOT_COMPILE oggvorbis_artsplugin"
 fi
--- origsrc/kdemultimedia-3.5.9/xine_artsplugin/Makefile.am	2006-07-22 03:10:54.000000000 -0500
+++ src/kdemultimedia-3.5.9/xine_artsplugin/Makefile.am	2008-04-29 20:56:08.921875000 -0500
@@ -7,7 +7,7 @@
 libarts_xine_la_SOURCES = xinePlayObject.cc \
 			  xinePlayObject_impl.cpp \
 			  audio_fifo_out.c
-libarts_xine_la_LDFLAGS = $(all_libraries) -module -no-undefined -pthread
+libarts_xine_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 libarts_xine_la_LIBADD = $(XINE_LIBS) $(LIB_X11) $(LIB_XEXT) \
 			 -lkmedia2_idl -lsoundserver_idl -lartsflow
 
