--- origsrc/kdemultimedia-4.0.99/kmix/CMakeLists.txt	2008-07-08 04:54:47.000000000 -0500
+++ src/kdemultimedia-4.0.99/kmix/CMakeLists.txt	2008-07-24 11:43:46.546875000 -0500
@@ -56,7 +56,7 @@
     target_link_libraries(kdeinit_kmix ${ASOUND_LIBRARY})
 endif (HAVE_LIBASOUND2)
 
-install(TARGETS kdeinit_kmix  DESTINATION ${LIB_INSTALL_DIR} )
+install(TARGETS kdeinit_kmix  DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 target_link_libraries( kmix kdeinit_kmix )
 install(TARGETS kmix  ${INSTALL_TARGETS_DEFAULT_ARGS} )
@@ -84,7 +84,7 @@
     target_link_libraries(kdeinit_kmixctrl ${ASOUND_LIBRARY})
 endif (HAVE_LIBASOUND2)
 
-install(TARGETS kdeinit_kmixctrl  DESTINATION ${LIB_INSTALL_DIR} )
+install(TARGETS kdeinit_kmixctrl  DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 target_link_libraries( kmixctrl kdeinit_kmixctrl )
 install(TARGETS kmixctrl ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- origsrc/kdemultimedia-4.0.99/libkcompactdisc/CMakeLists.txt	2008-07-08 04:54:46.000000000 -0500
+++ src/kdemultimedia-4.0.99/libkcompactdisc/CMakeLists.txt	2008-07-24 11:16:55.312500000 -0500
@@ -32,11 +32,11 @@
         wmlib/drv_toshiba.c
 )
 
-if (APPLE OR WIN32)
+if (APPLE OR WIN32 OR CYGWIN)
 	set(USE_WMLIB false)
-else (APPLE OR WIN32)
+else (APPLE OR WIN32 OR CYGWIN)
 	set(USE_WMLIB true)
-endif (APPLE OR WIN32)
+endif (APPLE OR WIN32 OR CYGWIN)
 
 set(kcompactdisc_LIB_SRCS kcompactdisc.cpp kcompactdisc_p.cpp phonon_interface.cpp)
 
